source/ConditionsStore.cpp

```diff
@@ -0,0 +1,131 @@
+/* ConditionsStore.cpp
+Copyright (c) 2020 by Peter van der Meer
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "ConditionsStore.h"
+
+using namespace std;
+
+
+
+// Empty constructor, just results in an emtpy store.
+ConditionsStore::ConditionsStore()
+{
+}
+
+
+
+// Constructor where a number of initial manually-set values are set.
+ConditionsStore::ConditionsStore(initializer_list<pair<string, int64_t>> initialConditions)
+{
+	for(auto it : initialConditions)
+		SetCondition(it.first, it.second);
+}
+
+
+
+// Constructor where a number of initial manually-set values are set.
+ConditionsStore::ConditionsStore(const map<string, int64_t> initialConditions)
+{
+	for(auto it : initialConditions)
+		SetCondition(it.first, it.second);
+}
+
+
+
+int64_t ConditionsStore::operator [] (const std::string &name) const
+{
+	return GetCondition(name);
+}
+
+
+
+bool ConditionsStore::HasCondition(const std::string &name) const
+{
+	// When we add support for on-demand conditions, then we can
+	// lookup the on-demand condition here before searching the
+	// manually set conditions.
```

Comment:
Looking forward to this. :)

Reply:
Yes, the end-goal of what this PR is preparing for. :-)